pkg/route/api/v1beta3/types.go

```diff
@@ -0,0 +1,104 @@
+package v1beta3
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// Route encapsulates the inputs needed to connect an alias to endpoints.
+type Route struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RouteSpec   `json:"spec"`
+	Status RouteStatus `json:"status"`
+}
+
+// RouteList is a collection of Routes.
+type RouteList struct {
+	kapi.TypeMeta `json:",inline"`
+	kapi.ListMeta `json:"metadata,omitempty"`
+	Items         []Route `json:"items"`
+}
+
+// RouteSpec describes the route the user wishes to exist.
+type RouteSpec struct {
+	// Ports are the ports that the user wishes to expose.
+	//Ports []RoutePort `json:"ports,omitempty"`
+
+	// Optional: Alias/DNS that points to the service
+	// Can be host or host:port
+	// host and port are combined to follow the net/url URL struct
+	Host string `json:"host"`
+	// Optional: Path that the router watches for, to route traffic for to the service
+	Path string `json:"path,omitempty"`
+
+	// An object the route points to. Only the Service kind is allowed, and it will
+	// be defaulted to Service.
+	To kapi.ObjectReference `json:"to"`
+
+	// TLS provides the ability to configure certificates and termination for the route
+	TLS *TLSConfig `json:"tls,omitempty"`
+}
+
+/*
+type RoutePort struct {
+	// Name is the name of the port that is used by the router. Routers may require
+	// this field be set. Routers may decide which names to expose.
+	Name string `json:"name"`
+
+	// Optional: the name of the target endpoint port.
+	TargetName string `json:"targetName"`
+
+	// Optional: the value of the target endpoint port to expose. May be omitted if
+	// name is set, and vice versa.
+	TargetPort util.IntOrString `json:"targetPort"`
+}
+*/
+
+// RouteStatus describes the current state of this route.
+type RouteStatus struct{}
```

Comment:
Until we know how the shard is going to be organized (will we do Spec.Shard like we do for Pods, or Spec.Shards[]) I want to hold off.

----- Original Message -----
> > +type RoutePort struct {
> > +	// Name is the name of the port that is used by the router. Routers may
> > require
> > +	// this field be set. Routers may decide which names to expose.
> > +	Name string `json:"name"`
> > +
> > +	// Optional: the name of the target endpoint port.
> > +	TargetName string `json:"targetName"`
> > +
> > +	// Optional: the value of the target endpoint port to expose. May be
> > omitted if
> > +	// name is set, and vice versa.
> > +	TargetPort util.IntOrString `json:"targetPort"`
> > +}
> > +*/
> > +
> > +// RouteStatus describes the current state of this route.
> > +type RouteStatus struct{}
> 
> We can probably go ahead and add a reference to the RouterShard here unless
> the plan is to hold it in an annotation.
> 
> Statuses that I can think of: Pending, Allocated, Accepted,
> Running/Deployed/Live, Failed
> 
> Accepted would be a short transition state where the router has received the
> route but has not yet written config/recycled the router.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1906/files#r29071514